src/corona_hakab_model/healthcare.py

```diff
@@ -0,0 +1,109 @@
+from __future__ import annotations
+
+from collections import namedtuple
+from typing import TYPE_CHECKING, Callable, List
+
+import numpy as np
+from agent import Agent
+from util import Queue
+
+if TYPE_CHECKING:
+    from manager import SimulationManager
+
+
+PendingTestResult = namedtuple("PendingTestResult", ["agent", "test_result", "original_duration"])
+
+
+class PendingTestResults(Queue[PendingTestResult]):
+    def __init__(self):
+        super().__init__()
+
+
+class DetectionTest:
+    def __init__(self, detection_prob, false_alarm_prob, time_until_result):
+        self.detection_prob = detection_prob
+        self.false_alarm_prob = false_alarm_prob
+        self.time_until_result = time_until_result
+
+    def test(self, agent: Agent):
+        if agent.medical_state.detectable:
+            test_result = np.random.rand() < self.detection_prob
+        else:
+            test_result = np.random.rand() < self.false_alarm_prob
+
+        pending_result = PendingTestResult(agent, test_result, self.time_until_result)
+        return pending_result
+
+
+class HealthcareManager:
+    def __init__(self, sim_manager: SimulationManager):
+        self.manager = sim_manager
+
+    def _get_testable(self):
+        tested_pos_too_recently = (
+            self.manager.tested_vector
+            & self.manager.tested_positive_vector
+            & (
+                self.manager.current_date - self.manager.date_of_last_test
+                < self.manager.consts.testing_gap_after_positive_test
+            )
+        )
+
+        tested_neg_too_recently = (
+            self.manager.tested_vector
+            & np.logical_not(self.manager.tested_positive_vector)
+            & (
+                self.manager.current_date - self.manager.date_of_last_test
+                < self.manager.consts.testing_gap_after_negative_test
+            )
+        )
+
+        return np.logical_not(tested_pos_too_recently | tested_neg_too_recently) & self.manager.living_agents_vector
+
+    def testing_step(self, test: DetectionTest, num_of_tests, priorities: List[Callable[[Agent], bool]]):
```

Comment:
Consider splitting this method into smaller functions in the future.
not critical for this PR :)